### DISCUSSION TITLE
Model improvement and project betterment brainstorming

### DISCUSSION THREAD

**Discussion by bhupali08** [bhupali08/Image-Classification-using-Deep-Learning-for-a-city-dog-show#11] [Q&A] [answered]
The goal of starting this discussion thread is to expand on this project and try to make the project better, explore its transferability into another domain, and test the possibilities of deploying an application. 

Hence, some of the questions that I want to brainstorm are: 
1. how can the model be further optimized?
2. how can the model decisions be better interpreted? 
3. error analysis of the model to understand where it is lacking.

**Comment by SohamBera16** [comment, marked as answer]
1. Some of the ways to further optimize the model performance could be to tune the hyperparameters (using Grid Search or Bayesian Optimization based search) and use regularization methods like early stopping, dropout, etc. Also, data augmentation techniques could be explored for better performance. 
2. To interpret what drives the model decisions, we can perhaps use the Class Activation Map (CAM) approach so that we can visualize the neuron as well as layer-level features that the Neural Network extracts and try to interpret which features are influencing a certain classification for an image and whether those features can be made more prominent so that the model learns them even more effectively. 
3. We can see the samples on which the model misclassifies, then try to perform error analysis and figure out what confuses the model to make a wrong decision. Accordingly, we can make the dataset or the model architecture even better.
